web/packages/teleport/src/Navigation/SideNavigation/Navigation.tsx

```diff
@@ -325,114 +395,131 @@ export function Navigation() {
+      onKeyUp={e => e.key === 'Escape' && resetExpandedSection(false)}
       onBlur={(event: React.FocusEvent<HTMLDivElement, Element>) => {
         if (!event.currentTarget.contains(event.relatedTarget)) {
           resetExpandedSection();
         }
       }}
-      css={`
-        position: relative;
-        width: var(--sidenav-width);
-        z-index: ${zIndexMap.sideNavContainer};
-      `}
+      className={
+        stickyMode &&
+        currentPageSection &&
+        !!debouncedSection &&
+        !debouncedSection.standalone
+          ? 'sticky-mode'
+          : ''
+      }
     >
       <SideNavContainer>
         <PanelBackground />
         <SearchSection
-          navigationSections={[...navSections, topMenuSection]}
+          navigationSections={[...combinedSideNavSections, topMenuSection]}
           expandedSection={debouncedSection}
           previousExpandedSection={previousExpandedSection}
           handleSetExpandedSection={handleSetExpandedSection}
           currentView={currentView}
+          stickyMode={stickyMode}
+          toggleStickyMode={toggleStickyMode}
+          canToggleStickyMode={!!currentPageSection}
         />
         <ResourcesSection
           expandedSection={debouncedSection}
           previousExpandedSection={previousExpandedSection}
           handleSetExpandedSection={handleSetExpandedSection}
           currentView={currentView}
+          stickyMode={stickyMode}
+          toggleStickyMode={toggleStickyMode}
+          canToggleStickyMode={!!currentPageSection}
         />
-        {navSections.map(section => (
-          <React.Fragment key={section.category}>
-            {section.category === 'Add New' && <Divider />}
-            <Section
-              key={section.category}
-              section={section}
-              $active={section.category === currentView?.category}
-              setExpandedSection={() => handleSetExpandedSection(section)}
-              aria-controls={`panel-${debouncedSection?.category}`}
-              onClick={() => {
-                if (section.standalone) {
-                  handleNavigation(section.subsections[0].route);
-                }
-              }}
-              isExpanded={
-                !!debouncedSection &&
-                !debouncedSection.standalone &&
-                section.category === debouncedSection?.category
-              }
-            >
-              <RightPanel
-                isVisible={
-                  !!debouncedSection &&
-                  !debouncedSection.standalone &&
-                  section.category === debouncedSection?.category
-                }
-                skipAnimation={!!previousExpandedSection}
-                id={`panel-${section.category}`}
-                onFocus={() => handleSetExpandedSection(section)}
-                onMouseEnter={() => handleSetExpandedSection(section)}
+        {navSections.map(section => {
```

Comment:
This is a pretty big section to have inside the returned JSX - can this be another component?